torch/_dynamo/guards.py

```diff
@@ -394,6 +394,8 @@ def EQUALS_MATCH(self, guard: Guard):
             torch.Size,
             torch.device,
             torch.dtype,
+            torch.memory_format,
+            torch.layout,
```

Comment:
There are at least three places referring the common constant types list, we should combine them together. But currently they are not identical, I think they should be. I'll leave it as a separate PR in case it triggers unexpected issues.